vllm/v1/engine/core.py

```diff
@@ -1236,6 +1243,14 @@ def run_busy_loop(self):
                 # if the model didn't execute any ready requests.
                 self.execute_dummy_batch()
 
+                # After dummy batch, check if we should continue or wait.
+                # If we have no local work, sync with other ranks before continuing.
+                if not local_unfinished_reqs:
+                    self.engines_running = self._has_global_unfinished_reqs(
+                        local_unfinished_reqs
+                    )
+                    continue
```

Comment:
By adding this you're skipping the wave count increment + step counter reset below. Could you explain why this is necessary in this case ?
Maybe removing this code and simply adding
```
if not executed and not local_unfinished_reqs:
    continue
```
just below in step 3) is cleaner/clearer.